test/cli/credentials/list.test.js

```diff
@@ -1,58 +1,187 @@
 const assert = require('assert');
+const idGen = require('uuid-base62');
 const adminHelper = require('../../common/admin-helper')();
 const environment = require('../../fixtures/cli/environment');
 const namespace = 'express-gateway:credentials:list';
-const idGen = require('uuid-base62');
 
 describe('eg credentials list -c ', () => {
-  let program, env, user, keyCred1;
+  let program, env, username;
+
+  const createdTypes = {
+    inc (type, isActive) {
+      if (isActive) {
+        this.active[type] = (this.active[type] || 0) + 1;
+      } else {
+        this.deactivated[type] = (this.deactivated[type] || 0) + 1;
+      }
+      this.all[type] = (this.all[type] || 0) + 1;
+    },
+    reset () {
+      this.active = {};
+      this.deactivated = {};
+      this.all = {};
+    }
+  };
+
+  const createdKeyAuthKeys = {
+    add (keyId, isActive) {
+      if (isActive) {
+        this.active.push(keyId);
+      } else {
+        this.deactivated.push(keyId);
+      }
+      this.all.push(keyId);
+    },
+    reset () {
+      this.active = [];
+      this.deactivated = [];
+      this.all = [];
+    }
+  };
+
+  const createCredential = (type, options = {}, isActive = true) => {
+    const {credentials} = adminHelper.admin;
+    return credentials
+      .create(username, type, options)
+      .then((credential) => {
+        const promises = [];
+        createdTypes.inc(type, isActive);
+
+        switch (type) {
+          case 'key-auth':
+            const {keyId} = credential;
+            createdKeyAuthKeys.add(keyId, isActive);
+            if (!isActive) {
+              promises.push(credentials.deactivate(keyId, type));
+            }
+            break;
+        }
+
+        return Promise.all(promises);
+      });
+  };
+
   before(() => {
-    ({ program, env } = environment.bootstrap());
+    ({program, env} = environment.bootstrap());
     return adminHelper.start();
   });
+
   after(() => adminHelper.stop());
 
   beforeEach(() => {
+    createdTypes.reset();
+    createdKeyAuthKeys.reset();
+
     env.prepareHijack();
-    return adminHelper.admin.users.create({
-      username: idGen.v4(),
-      firstname: 'La',
-      lastname: 'Deeda'
-    })
-    .then(createdUser => {
-      user = createdUser;
-      return Promise.all([
-        adminHelper.admin.credentials.create(user.username, 'key-auth', {}),
-        adminHelper.admin.credentials.create(user.username, 'basic-auth', {password: 'test'}),
-        adminHelper.admin.credentials.create(user.username, 'oauth2', {secret: 'eg'})
-      ]);
-    })
-    .then(([keyCred1Res, keyCred2Res, basicCredRes, oauth2CredRes]) => {
-      keyCred1 = keyCred1Res;
+
+    return adminHelper
+      .admin
```

Comment:
just curios how it is better from `adminHelper.admin.users.create({  ` ?

Reply:
?

Reply:
I see new lines, just trying to understand style preference

Reply:
In my opinion it's better to read.

Reply:
I agree